### PR TITLE
chore(build): add automatic release

### LINKED ISSUE #74: Look at adding semantic-release to CI to automate versioning/releases
https://github.com/semantic-release/semantic-release

### PR DESCRIPTION
**Pull request by devpow112**
This adds `semantic-release` to handle automatic releases as well as `commitlint` to enforce commit message styling. This will generate a new release once a day or immediately on bug fixes. An automatic release will create a new git tag as well as create a github release that will look something like this https://github.com/devpow112/serverless-plugin-function-value/releases/tag/v1.1.0. In order to have this work fully I have changed from merge to rebase for PR's and required linear history. This will also help with `master` always being something that works. This resolves #74.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,27 +1,45 @@
...
 jobs:
   build:
     name: build
+    if: (!startsWith(github.event.head_commit.message, 'chore(release):'))
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: Brightspace/third-party-actions@actions/checkout
+        with:
+          fetch-depth: 0
```
Comment (cbelan-d2l) [inline .github/workflows/build.yml:18]: This is to get the full history of the branch

File: .releaserc.json [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+						"type": "refactor",
+						"release": "patch"
+					}
+				]
+			}
+		],
+		[
+			"@semantic-release/github",
+			{
+				"labels": [
```
Comment (cbelan-d2l) [inline .releaserc.json:26]: If release fails, it will create a defect and add a label `auto`

File: .releaserc.json [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+					"auto"
+				],
+				"releasedLabels": [
+					"released"
```
Comment (cbelan-d2l) [inline .releaserc.json:30]: Once something is released, the PR will get this label

File: .releaserc.json [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+							"section": "Build System"
+						}
+					]
+				}
+			}
+		],
+		[
+			"@semantic-release/git",
+			{
+				"assets": [
```
Comment (cbelan-d2l) [inline .releaserc.json:89]: Commit in master

**Review by cbelan-d2l — APPROVED** [review]
Looks great!

Thanks for running through the code with me 🎉

File: .github/workflows/build.yml (hunk 1 above)
Comment (devpow112) [reply .github/workflows/build.yml:18]: Looks like I will need a PAT from `brightspace-bot` to make this work as hoped https://github.com/semantic-release/semantic-release/blob/master/docs/recipes/github-actions.md#pushing-packagejson-changes-to-a-master-branch

**Comment by devpow112** [comment]
Closing for now, will re-open if this ever becomes a thing again.

**Pull request closed without merging**